Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -357,9 +337,9 @@ private bool InstallSourceKnown(string url)
             return url.StartsWith(acceptedSource) && Context.API.GetAllPlugins().Any(x => x.Metadata.Website.StartsWith(contructedUrlPart));
         }
 
-        internal async ValueTask<List<Result>> RequestInstallOrUpdate(string search, CancellationToken token)
+        internal async ValueTask<List<Result>> RequestInstallOrUpdate(string search, CancellationToken token, bool usePrimaryUrlOnly = false)
         {
-            await UpdateManifestAsync(token);
+            await PluginsManifest.UpdateManifestAsync(token, usePrimaryUrlOnly);
```

Comment:
If we fetch result every time, the ctrl+r will only refresh the result on the first one, but later one will still use the manifest in cdn? I believe we should have local cache for this.

Reply:
We don't fetch _every time_, I have added a 10-second timeout so that we don't fetch repeatedly while a user is typing. Hitting `Ctrl+R` will bypass the timeout and fetch from the primary url.

> Inserting another character will revert back to the old results

so this is only an issue if the user waits 10 seconds before inserting that extra character.

Reply:
10 seconds seems to be pretty short? As we have the ctrl+r features, why don't we add a couple minutes timeout?

Reply:
> 10 seconds seems to be pretty short? 

it's only meant to avoid multiple http requests while a user is typing

> As we have the ctrl+r features, why don't we add a couple minutes timeout?

I don't see a reason to: fetching is very fast with the CDN links, and I've added 3 CDN links now so that fetching should be fast for users anywhere in the world (the fastest cdn link will be used in each user's case).

It takes around 80ms for me, a 2-minute timeout for a call this quick seems a bit unreasonable 😂

Reply:
If you are in China, the cdn is not so fast. It takes around 1s to download the manifest for me without proxy (with proxy it shall be fast).

![image](https://github.com/Flow-Launcher/Flow.Launcher/assets/45326534/94c7cb03-d469-4fbb-9fe5-2337db933c42)

Reply:
Okay, I can increase the timeout to a couple minutes. Does the rest of the PR look good to you?

> the other cdn takes 600ms, but sometimes it doesn't work.

yeah that's served by cloudflare, don't think they're very solid in China

Reply:
The rest looks good to me. I will take a detail look today. Previously jsdelivr has license in China, so they are able to deliver cdn that located in China, which is much much faster and stabler.